### PR TITLE
[#120100877] Manage pingdom contacts

### PR DESCRIPTION
**Pull request by saliceti**
# What

Story: [Email Alerts from Pingdom](https://www.pivotaltracker.com/story/show/120100877)

We want to be able to set pingdom notification alerts via [terraform](https://github.com/alphagov/paas-terraform-provider-pingdom). This requires managing contact IDs on the checks. We are adding this feature to the library.

The first intent of the story was to manage contacts end-to-end but this will be for another story now. The code is functional so it can be merged too.
# How to review

This can be tested independently. The PR on the terraform provider will be tested including this library.
This requires a pro account such as the one the team has.
You can play with this example code for example:

``` go
    client := pingdom.NewMultiUserClient("username","password", "app key", "account email")
    url, _ := url.Parse("https://api.pingdom.com")
    client.BaseURL = url

    // List contacts
    contacts, err := client.Contacts.List()
    if err != nil {
        fmt.Println(err)
    }
    fmt.Println(contacts)

    // Create contact
    newContact := pingdom.Contact{
        Name:               "Test " + strconv.Itoa(rand.Int()),
        Email:              "test@site.com",
        Defaultsmsprovider: "nexmo",
    }

    contactResponse, err := client.Contacts.Create(&newContact)
    if err != nil {
        fmt.Println(err)
    }

    contactID := contactResponse.ID

    contacts, err = client.Contacts.List()
    if err != nil {
        fmt.Println(err)
    }
    fmt.Println(contacts)

    // Create check with contact ID
    newCheck := pingdom.HttpCheck{
        Name:                     "HTTP check " + strconv.Itoa(rand.Int()),
        Hostname:                 "example.com",
        Resolution:               1,
        ContactIds:               []int{contactID},
        SendNotificationWhenDown: 2,
    }
    checkResponse, err := client.Checks.Create(&newCheck)
    if err != nil {
        fmt.Println(err)
    }

    check, err := client.Checks.Read(checkResponse.ID)
    if err != nil {
        fmt.Println(err)
    }
    fmt.Println(check)

    _, err = client.Checks.Delete(checkResponse.ID)
    if err != nil {
        fmt.Println(err)
    }
    // Update contact
    updateContact := pingdom.Contact{
        Name:               "Test " + strconv.Itoa(rand.Int()),
        Email:              "test@site.com",
        Defaultsmsprovider: "nexmo",
    }
    _, err = client.Contacts.Update(contactID, &updateContact)
    if err != nil {
        fmt.Println(err)
    }

    contacts, err = client.Contacts.List()
    if err != nil {
        fmt.Println(err)
    }
    fmt.Println(contacts)

    // Delete contact
    _, err = client.Contacts.Delete(contactResponse.ID)
    if err != nil {
        fmt.Println(err)
    }

    contacts, err = client.Contacts.List()
    if err != nil {
        fmt.Println(err)
    }
    fmt.Println(contacts)
```
# Who can review

Not myself, @HenryTK or @combor

### CODE REVIEW

File: pingdom/contact_types.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	Cellphone          string            `json:"cellphone,omitempty"`
+	Countryiso         string            `json:"countryiso,omitempty"`
+	CountryCode        string            `json:"countrycode,omitempty"`
+	Defaultsmsprovider string            `json:"defaultsmsprovider,omitempty"`
+	Directtwitter      bool              `json:"directtwitter,omitempty"`
+	Twitteruser        string            `json:"twitteruser,omitempty"`
+	Iphonetokens       string            `json:"iphonetokens,omitempty"`
+	Androidtokens      string            `json:"androidtokens,omitempty"`
+	Paused             bool              `json:"paused,omitempty"`
+	RequestHeaders     map[string]string `json:"requestheaders,omitempty"`
```
Comment (keymon) [inline pingdom/contact_types.go]: What is this requestheaders field? I do not see in the struct of the contact, and it does not seem to be related to a contact at all. 

Maybe we added this by accident?

Comment (saliceti) [inline pingdom/contact_types.go]: You're right. I think it comes from checks.
I'll remove it.

File: README.md [hunk 2]
```diff
@@ -84,3 +84,16 @@ Delete a check:
...
 ```
 
+Create a notification contact (multi-user accounts):
+
+```go
+newContact := pingdom.Contact{ Name: "John Doe", Email: "john.doe@example.com", Defaultsmsprovider: "nexmo" }
+contactResponse, err := client.Contacts.Create(&newContact)
+```
+
+Create a check with basic alert notification to a contact:
```
Comment (keymon) [inline README.md]: I think we should highlight that you must set:

```
    SendNotificationWhenDown: 2,
    SendToEmail: true, 
    UseLegacyNotifications: true,

```

**Comment by keymon** [comment]
LGTM, will merge as soon as you resolve the two comments

**Comment by keymon** [comment]
+1

**Comment by keymon** [comment]
Branch left to push upstream
